### PR TITLE
Change license, flesh out README a bit.

### PR DESCRIPTION
**Pull request by c0gent**
* Expand readme
* Change license

Closes https://github.com/poanetwork/threshold_crypto/issues/7 and https://github.com/poanetwork/threshold_crypto/issues/2

@andogro, I'll hand this off to you to refine the language in a separate PR as you see fit.

Many of the tests use private types or types with private constructors. I'll leave it to @vkomenda, @afck, and @DrPeterVanNostrand to build on the example and decide what you may want to make public for those purposes.

For now this is probably ready to merge.

### CODE REVIEW

**Comment by c0gent** [comment]
Also, we should probably re-export `rand` since we're using an old version not to mention the fact that it's basically required.

File: COPYRIGHT [hunk 1]
```diff
@@ -0,0 +1,7 @@
+threshold_crypto is copyright 2018, POA Network Developers.
+
+Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+http://www.apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+http://opensource.org/licenses/MIT>, at your option. All files in the project
+carrying such notice may not be copied, modified, or distributed except
+according to those terms.
```
Comment (vkomenda) [inline COPYRIGHT]: Can you add a newline here?

File: COPYRIGHT [hunk 2]
```diff
@@ -0,0 +1,7 @@
+threshold_crypto is copyright 2018, POA Network Developers.
```
Comment (vkomenda) [inline COPYRIGHT]: Do you think "POA Network" could also be OK? In addition, POA Network (Ltd) is a legal entity. Although I don't think we should write "POA Network Ltd."

File: LICENSE-MIT [hunk 3]
```diff
@@ -0,0 +1,21 @@
+The MIT License
+
+Copyright (c) 2018 POA Network Developers
```
Comment (vkomenda) [inline LICENSE-MIT]: I'd just say "POA Network" here.

File: COPYRIGHT (hunk 2 above)
Comment (c0gent) [reply COPYRIGHT]: Yeah, we can put whatever there. @igorbarinov, what should we put?

File: COPYRIGHT (hunk 1 above)
Comment (c0gent) [reply COPYRIGHT]: Sorry, a newline where@

**Comment by c0gent** [comment]
I suppose that the lawyers can give the documents a once over but it's pretty much boilerplate copy and paste :)

File: COPYRIGHT (hunk 1 above)
Comment (vkomenda) [reply COPYRIGHT]: Just after the last sencence. Github displays a warning about the missing newline at the end of file.

File: README.md [hunk 4]
```diff
@@ -1,6 +1,101 @@
...
+
+Provides constructors for encrypted message handling within a public key
+encryption system. It utilizes the pairing elliptic curve library to create
+and enable reconstruction of public and private key shares.
+
+In a network environment, messages are signed and encrypted, and key and
+signature shares are distributed to network participants. A message can be
+decrypted and authenticated only with cooperation from at least `threshold +
+1` nodes.
+
```
[the remark this answers is not in the document]
Comment (c0gent) [reply README.md]: No, that's just rustdoc or internal doc material.

File: COPYRIGHT (hunk 1 above)
Comment (c0gent) [reply COPYRIGHT]: It's not code so don't worry about it :)

File: COPYRIGHT (hunk 2 above)
Comment (igorbarinov) [reply COPYRIGHT]: Why not use the company name? e.g. Facebook MIT license uses Facebook Inc. https://github.com/facebook/flow/blob/master/LICENSE as a grantor of the license. If it's ok let's use `POA Networks Ltd.`

**Comment by vkomenda** [comment]
@c0gent, are you still planning to re-export `rand`? If not, this PR can be merged.

**Pull request merged**